wrap/aci/v1/handler.go

```diff
@@ -75,27 +76,25 @@ func (h *Handler) Run() {
 	h.InitHandler()
 
 	go func() {
-		iter := 0
+		nodes := h.Config.ACIHosts
 		for {
 			if h.Config.ConsulEnabled {
-				nodes := h.GetConsulNodes()
-				h.Log.Debugf("ACI nodes received from consul: %s\n", nodes)
-				n := len(nodes)
-
-				if n != 0 {
-					// Round-Robin over the nodes
-					if iter >= n {
-						iter = 0
-					}
-					h.Config.SourceHostname = nodes[iter%n]
-					iter += 1
-				} else {
-					h.Log.Fatalf("No nodes could be found in consul. Consul config: %s", h.sc.Consul)
-				}
+				nodes = h.GetConsulNodes()
+				h.Log.Infof("ACI nodes received from consul: %s\n", nodes)
+			}
+			if len(nodes) <= 0 {
+				h.Log.Infof("Empty host list. Sleeping for %d seconds.\n", h.Config.CycleInterval)
+				time.Sleep(time.Duration(h.Config.CycleInterval) * time.Second)
+				continue
+			}
+			sleepTime := h.Config.CycleInterval / len(nodes)
```

Comment:
Maybe : Measure time taken for total run and if time taken is less than `CycleInterval`, sleep for `CycleInterval - timeTaken`, else don't sleep.

Reply:
I like your suggestion, I will change it